packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1299,13 +1299,86 @@ class RenderFractionalTranslation extends RenderProxyBox {
   }
 }
 
+/// The interface used by [CustomPaint] (in the widgets library) and
+/// [RenderCustomPaint] (in the rendering library).
+///
+/// To implement a custom painter, subclass this interface to define your custom
+/// paint delegate.
+///
+/// The [paint] method is called whenever the custom object needs to be
+/// repainted. This is the primary method that must be implemented.
+///
+/// The [shouldRepaint] method must also be implemented. It is called when a new
+/// instance of a delegate class is provided, to check if the new instance
+/// actually represents different information.
+///
+/// The [hitTest] method can optionally be implemented also.
 abstract class CustomPainter {
   /// Abstract const constructor. This constructor enables subclasses to provide
   /// const constructors so that they can be used in const expressions.
   const CustomPainter();
 
+  /// Called whenever the object needs to paint. The given [Canvas] has its
+  /// coordinate space configured such that the origin is at the top left of the
+  /// box. The area of the box is the size of the [size] argument.
+  ///
+  /// Paint operations should remain inside the given area. Graphical operations
+  /// outside the bounds may be silently ignored, clipped, or not clipped; no
+  /// guarantees are made.
+  ///
+  /// Implementations should be wary of correctly pairing any calls to
+  /// [Canvas.save]/[Canvas.saveLayer] and [Canvas.restore], otherwise all
+  /// subsequent painting on this canvas may be affected, with potentially
+  /// hilarious but confusing results.
+  ///
+  /// To paint text on a [Canvas], use a [TextPainter].
+  ///
+  /// To paint an image, obtain an [ImageResource] (e.g. using the
+  /// [ImageCache.load] method on the [imageCache] singleton), and create a new
+  /// instance of your custom delegate, giving it the [ImageResource] object's
+  /// [ImageInfo] object whenever it changes. Then, inside your delegate's
+  /// [paint] method, call the [Canvas.drawImage], [Canvas.drawImageRect], or
+  /// [Canvas.drawImageNine] methods to paint the [ImageInfo.image] object,
+  /// applying the [ImageInfo.scale] value to obtain the correct rendering size.
   void paint(Canvas canvas, Size size);
+
+  /// Called whenever a new instance of the custom painter delegate class is
+  /// provided to the [RenderCustomPaint] object, or any time that a new
+  /// [CustomPaint] object is created with a new instance of the custom painter
+  /// delegate class (which amounts to the same thing, since the latter is
+  /// implemented in terms of the former).
+  ///
+  /// If the new instance represents different information than the old
+  /// instance, then the method should return true, otherwise it should return
+  /// false.
+  ///
+  /// If the method returns false, then the paint call might be optimised away.
```

Comment:
I mean "might".

"can" implies either that the author is able to do it (which isn't the case, it's up to the framework), or that it is always possible for the framework to do so in that condition (which it usually isn't). "might" implies that there is a chance that it will happen, but that it's not a given.